rio/api/src/main/java/org/eclipse/rdf4j/rio/RioConfig.java

```diff
@@ -112,4 +116,15 @@ public RioConfig useDefaults() {
 
 		return this;
 	}
+
+	/**
+	 * Checks if the supplied {@link RioSetting}'s default value has been override by setting a system property.
+	 * 
+	 * @param setting
+	 *        the setting to check the existence of an override for
+	 * @return true if the default value is overridden by system property, false otherwise.
+	 */
+	private boolean hasSystemPropertyOverride(RioSetting<?> setting) {
+		return !Objects.isNull(System.getProperty(setting.getKey()));
```

Comment:
This still has the issue that it can go out of sync with the actual setting default. If the system property is set just before RDF4J is classloaded, the `RioSetting` default may be set (if/when it is accessed), but if the system property is then unset, this will return false and the default will not be used, which goes against intuition for how it should function if typical system properties are being supported. Attempting to add tests (which haven't been done so far in this PR)

The overall approach of changing static defaults based on mutable system properties is not going to be intuitive to most people due to the trickiness of having to set the system property before RDF4J is class loaded to get the sequence correct, and then any future changes will not affect the actual property, but will affect the result of calls to ``isSet`` as this uses the current property.

Part of the difficulty is trying to attach this without redesigning the overall workflow for RDFParser or RDFWriter, which has been stable since Sesame-2. It should be possible, but we need to ensure that the workflow stays intuitive and follows the principle of least surprise for the user.

An alternative to the approach implemented here could be to internally modify the well encapsulated internals of `RioConfig.get` to transparently look up the system property using a separate internal cache for the defaults. The new internal cache would be reset along with the existing settings in `useDefaults`.

This would work without modifying any `RioSetting`s. It also has the advantage of least surprise in that the user can setup the system property at any point before a `RDFParser`/`RDFWriter` run occurs, without having to presume that RDF4J wasn't classloaded before their user code is run for the first time. By not modifying the internals of `RioSetting` static instances it also has the possibility to be tweaked in future. The current approach here has no ability to be tweaked to work with long running JVMs or other user scenarios. 

It could also be useful to split out the key standardisation tweaks, which are not controversial other than that they break the public API by changing constant names (and hence will need to be targeted to a new major version), into a separate PR and merge it before focusing on support for system properties in a separate PR.

Reply:
> This still has the issue that it can go out of sync with the actual setting default. If the system property is set just before RDF4J is classloaded, the RioSetting default may be set (if/when it is accessed), but if the system property is then unset, this will return false and the default will not be used, which goes against intuition for how it should function if typical system properties are being supported

Good point.

>  Attempting to add tests (which haven't been done so far in this PR)

Not sure why you say that as this PR includes several new unit tests, though admittedly the case you sketch above is not covered.

I'll take a look at fixing along the lines you suggest, thanks again!

Reply:
Sorry about the tests comment, I meant that I would attempt to add tests if necessary to illustrate the usecase.